substrate/frame/support/procedural/src/pallet/parse/mod.rs

```diff
@@ -560,10 +563,35 @@ mod keyword {
 	syn::custom_keyword!(composite_enum);
 }
 
+/// The possible values for the `#[pallet::config]` attribute.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+enum ConfigValue {
+	/// `#[pallet::config(with_default)]`
+	WithDefault(keyword::with_default),
+	/// `#[pallet::config(without_metadata)]`
+	WithoutMetadata(keyword::without_metadata),
+}
+
+impl syn::parse::Parse for ConfigValue {
+	fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+		if input.peek(keyword::with_default) {
```

Comment:
You want to use this: https://docs.rs/syn/latest/syn/parse/struct.ParseBuffer.html#method.lookahead1